app/routes/cdo/create/microchip-search.js

```diff
@@ -53,19 +56,19 @@ module.exports = [{
         setDog(request, details)
       }
 
-      return h.redirect(results.length > 0 ? routes.microchipResults.get : routes.details.get)
+      return h.redirect(results.length > 0 ? `${routes.microchipResults.get}/${details.dogId}` : `${routes.details.get}/${details.dogId}`)
     }
   }
 }]
 
 const duplicateMicrochipsInSession = (request, details) => {
   const dogs = getDogs(request)
   if (dogs) {
-    dogs.forEach((d, ind) => { d.id = ind + 1 })
-    const dupeDogs = dogs.filter(d => d.microchipNumber === details.microchipNumber)
+    dogs.forEach((d, ind) => { d.dogId = d.dogId || `${ind + 1}` })
```

Comment:
is there a test to check for the `${d.dogId}` !== `${details.dogId}` case?

Reply:
Explicitly added now